mount/mount_linux.go

```diff
@@ -6,9 +6,47 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+const (
+	// ptypes is the set propagation types.
+	ptypes = unix.MS_SHARED | unix.MS_PRIVATE | unix.MS_SLAVE | unix.MS_UNBINDABLE
+
+	// pflags is the full set valid flags for a change propagation call.
+	pflags = ptypes | unix.MS_REC | unix.MS_SILENT
+
+	// broflags is the combination of bind and read only
+	broflags = unix.MS_BIND | unix.MS_RDONLY
+)
+
+// isremount returns true if either device name or flags identify a remount request, false otherwise.
+func isremount(device string, flags int) bool {
+	// We treat device "" and "none" as a remount request to provide compatibility with
+	// requests that don't explicitly set MS_REMOUNT such as those manipulating bind mounts.
+	return flags&unix.MS_REMOUNT != 0 || device == "" || device == "none"
```

Comment:
Also I think we should use the runc logic for flag checking here:

`flags&^(unix.MS_REC|unix.MS_REMOUNT|unix.MS_BIND) != 0`